config.go

```diff
@@ -419,36 +417,38 @@ func loadConfig() (*config, []string, error) {
 	// Update the home directory for dcrd if specified. Since the home
 	// directory is updated, other variables need to be updated to
 	// reflect the new changes.
-	if len(preCfg.DcrdHomeDir) > 0 {
-		cfg.DcrdHomeDir, _ = filepath.Abs(preCfg.DcrdHomeDir)
-		cfg.ConfigFile = filepath.Join(cfg.DcrdHomeDir, defaultConfigFilename)
-		cfg.DataDir = filepath.Join(cfg.DcrdHomeDir, defaultDataDirname)
-		cfg.RPCKey = filepath.Join(cfg.DcrdHomeDir, "rpc.key")
-		cfg.RPCCert = filepath.Join(cfg.DcrdHomeDir, "rpc.cert")
-		cfg.LogDir = filepath.Join(cfg.DcrdHomeDir, defaultLogDirname)
+	if len(preCfg.HomeDir) > 0 {
+		cfg.HomeDir, _ = filepath.Abs(preCfg.HomeDir)
+		cfg.ConfigFile = filepath.Join(cfg.HomeDir, defaultConfigFilename)
+		cfg.DataDir = filepath.Join(cfg.HomeDir, defaultDataDirname)
+		cfg.RPCKey = filepath.Join(cfg.HomeDir, "rpc.key")
+		cfg.RPCCert = filepath.Join(cfg.HomeDir, "rpc.cert")
+		cfg.LogDir = filepath.Join(cfg.HomeDir, defaultLogDirname)
 	}
 
 	// Load additional config from file.
+	var configFileError error
 	parser := newConfigParser(&cfg, &serviceOpts, flags.Default)
 	if !(preCfg.SimNet) || preCfg.ConfigFile !=
 		defaultConfigFile {
 
-		if _, err := os.Stat(preCfg.ConfigFile); os.IsNotExist(err) {
-			err := createDefaultConfigFile()
+		if _, err := os.Stat(cfg.ConfigFile); os.IsNotExist(err) {
```

Comment:
I don't believe this should have been changed.  Notice how the if statement above checks `preCfg.ConfigFile != defaultConfigFile`.